rdflib/plugins/serializers/turtle.py

```diff
@@ -224,7 +224,9 @@ def serialize(self, stream, base=None, encoding=None,
                   spacious=None, **args):
         self.reset()
         self.stream = stream
-        self.base = base
+        # if base is set here, override previously set base if set at graph init
+        if base is not None:
+            self.base = base
```

Comment:
In which case is `self.base` set already? The `TurtleSerializer` inherits from `RecursiveSerializer` from `Serializer` which is an `object`. It does not come from `Graph`.
The only reason is a previous call of `serialize`, which can be considered as a side effect.

Reply:
OK, yes I got that wrong, will change for this too. As above, I agree that setting base at serialization time should be for that serialization only - no side effects. An the Graph's base should override.

Reply:
I think the graphs base should only override in the case that the serialize base is None.

Reply:
Just check, this may not be arond the way you expect it to be.

Reply:
Just check this is now the way you expect it should be.